### DISCUSSION TITLE
IMPORT HISTORIC DATA INTO CURRENT BUDIBASE DATA STRUCTURE

### DISCUSSION THREAD

**Discussion by KernOil** [Budibase/budibase#1262] [Help]
Hi is there a way to import my historic data once I have built out the database in budibase?

**Comment by joebudi** [comment]
Hey! I'm not 100% sure what you mean, but you can continue to create new tables in your database and import data via a CSV, or add external data sources. 

What you cannot do, is create a table, add data, then at a later date, import more data to that table.

**Reply by KernOil** [reply]
gotcha, that last part was what I was wondering. Thank you. so if I build out the forms / pages I need in budibase before I load the historic data then I can no-longer add that historic data?

**Comment by louis-eyond** [comment]
Is there any plan to introduce an end-user import feature like this?  I am specifically interested in having the end user import a sales report from an external partner they receive on a monthly basis.
